### PR TITLE
Ci 15 nov 23

### PR DESCRIPTION
**Pull request by aimurphy**
* Some more TI revisions - [Generate the initial call graph](https://6555723e09555419410fc2c5--harness-developer.netlify.app/docs/continuous-integration/use-ci/run-tests/test-intelligence/ti-for-java-kotlin-scala#generate-the-initial-call-graph)
* Revise CI [Onboarding guide](https://6555723e09555419410fc2c5--harness-developer.netlify.app/docs/continuous-integration/get-started/onboarding-guide)
* Move and revise [Custom SCM status checks](https://6555723e09555419410fc2c5--harness-developer.netlify.app/docs/continuous-integration/use-ci/optimize-and-more/custom_github_status_check). Add step template and additional script examples.

### CODE REVIEW

**Comment by aimurphy** [comment]
@ravilach / @deepanshuHarness : Pranav requested that the custom SCM status checks article be moved out of the KB and into the main CI docs.

File: docs/continuous-integration/use-ci/optimize-and-more/custom_github_status_check.md
Comment (diegopereiraeng) [inline docs/continuous-integration/use-ci/optimize-and-more/custom_github_status_check.md:1]: i think the image is curlimages/curl:latest and not "curlimages/curl:lates"

**Review by diegopereiraeng — COMMENTED** [review]
Only the tag from the curl image should be changed

File: docs/continuous-integration/use-ci/optimize-and-more/custom_github_status_check.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+1. You need a [CI pipeline](../prep-ci-pipeline-components.md) with a [Build stage](../set-up-build-infrastructure/ci-stage-settings.md) where you'll add the **Run** step. If you haven't created a pipeline before, try one of the [CI pipeline tutorials](../../get-started/tutorials.md).
+2. Create [GitHub Personal Access Token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens) to use for authentication, and save the token as a [Harness text secret](/docs/platform/secrets/add-use-text-secrets).
+3. Add a [Run step](../run-ci-scripts/run-step-settings.md) to your **Build** (`CI`) stage.
+4. Enter a **Name** for the step.
+5. If required by your build infrastructure, provide the **Container Registry** and **Image**.
+   * **Container Registry** is a container registry connector, such as a Docker connector.
+   * **Image** is the FQN or artifact name and tag of a Docker image that has cURL installed, such as `curlimages/curl:lates` for the [official cURL Docker Hub image](https://hub.docker.com/r/curlimages/curl).
```
Comment (aimurphy) [inline docs/continuous-integration/use-ci/optimize-and-more/custom_github_status_check.md:21]:
```suggestion
   * **Image** is the FQN or artifact name and tag of a Docker image that has cURL installed, such as `curlimages/curl:latest` for the [official cURL Docker Hub image](https://hub.docker.com/r/curlimages/curl).
```

**Pull request merged**